vowpalwabbit/array_parameters.h

```diff
@@ -206,7 +207,8 @@ class sparse_parameters
     {
       for (auto& iter : _map)
       {
-        free(iter.second);
+        auto weight_ptr = iter.second;
```

Comment:
Nit: `auto*`

Reply:
is this for styling? because you don't need to declare pointers as `auto*`

Reply:
Just a preference thing, to me it makes it clear that this is a pointer so I don't need to worry about a potential copy here.